File: src/main/java/com/waitring/waitring/dto/user/UserInput.java
package com.waitring.waitring.dto.user;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.util.Date;

@Schema(description = "입력받은 회원 정보")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class UserInput {

    @Schema(description = "이메일")
    @NotBlank
    private String email;

    @Schema(description = "비밀번호")
    @NotBlank
    private String password;

    @Schema(description = "닉네임")
    @NotBlank
    private String nickname;

    @Schema(description = "포인트 점수")
    private Integer point;
}

File: src/main/java/com/waitring/waitring/dto/user/UserInput.kt
package com.waitring.waitring.dto.user

import io.swagger.v3.oas.annotations.media.Schema
import javax.validation.constraints.NotBlank

@Schema(description = "입력받은 회원 정보")
data class UserInput(
        @field:Schema(description = "이메일")
        @field:NotBlank
        val email: String?,

        @field:Schema(description = "비밀번호")
        @field:NotBlank
        val password: String?,

        @field:Schema(description = "닉네임")
        @field:NotBlank
        val nickname: String?,

        @field:Schema(description = "포인트 점수")
        val point: Int?,
)